plugins/chain_plugin/chain_plugin.cpp

```diff
@@ -868,7 +868,17 @@ void chain_plugin::plugin_initialize(const variables_map& options) {
       if(options.count( "block-log-retain-blocks" )) {
          my->chain_config->prune_config.emplace();
          my->chain_config->prune_config->prune_blocks = options.at( "block-log-retain-blocks" ).as<uint32_t>();
-         EOS_ASSERT(my->chain_config->prune_config->prune_blocks, plugin_config_exception, "block-log-retain-blocks cannot be 0");
+
+         if ( my->chain_config->prune_config->prune_blocks == 0 ) {
+            // clear out empty blocks.log. otherwise block_log::extract_genesis_state
+            // will return version 0 which asserts.
```

Comment:
How do you get into a state with a blocks.log of 0 size?

Reply:
I intentionally tested with blocks.log of 0 size. Probably it is not a valid test. Will revert this code.